### PR TITLE
nm.device: drop MTU reapply workaround support

### PR DESCRIPTION
**Pull request by ffmancera**
The minimum NetworkManager version supported is 1.20 so this patch drops
MTU reapply workaround support because it only applies for versions
lower than 1.18.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

File: libnmstate/nm/device.py [hunk 1]
```diff
@@ -187,45 +179,6 @@ def _modify_callback(src_object, result, user_data):
...
-def _mtu_changed(dev, connection_profile):
-    wired_setting = connection_profile.get_setting_wired()
-    configured_mtu = wired_setting.props.mtu if wired_setting else None
-    if configured_mtu:
-        current_mtu = int(dev.get_mtu())
-        return configured_mtu != current_mtu
-    return False
-
-
-def _ipv6_changed(dev, connection_profile):
```
Comment (tyll) [inline libnmstate/nm/device.py]: While it is correct to remove this, it does not belong into this commit, since it is a follow-up on #641

Comment (ffmancera) [reply libnmstate/nm/device.py]: Oops, yes you are right. Thanks!

Comment (ffmancera) [reply libnmstate/nm/device.py]: Fixed!

**Comment by ffmancera** [comment]
Oops I missed it. Sorry

> Uh, actually you seem to have removed too much compared to the previous release or I missed it, the call of `_requires_activation()` needs to be removed as well.

Oops, I missed it. Thanks!

**Pull request merged**